### PR TITLE
Battle Factory: Remove unnecessary set properties

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: data/random-teams.js [hunk 1]
```diff
@@ -1881,32 +1881,52 @@ class RandomTeams extends Dex.ModdedDex {
...
 			abilities.push(setData.abilityVariants ? randomAbility[setData.abilityVariants] : this.sample(randomAbility));
 		} else {
 			abilities.push(setData.set.ability);
 		}
 
+		let evs = {};
+		if (setData.set.evs) {
+			for (let ev in setData.set.evs) {
+				if (!setData.set.evs[ev]) evs[ev] = 0;
+				evs[ev] = setData.set.evs[ev];
```
Comment (urkerab) [inline data/random-teams.js]: Missing an `else`?

**Comment by Marty-D** [comment]
Nice! Could probably also throw out all the `"level" : 5`s in LC and just check if the tier is LC instead. BSS already has its own Factory anyway so everything else here will always be 100.

File: data/random-teams.js [hunk 2]
```diff
@@ -1881,32 +1881,52 @@ class RandomTeams extends Dex.ModdedDex {
...
+
+		let ivs = {};
+		if (setData.set.ivs) {
+			for (let iv in setData.set.ivs) {
+				if (!setData.set.ivs[iv]) ivs[iv] = 31;
+				ivs[iv] = setData.set.ivs[iv];
+			}
+		} else {
+			ivs = {hp: 31, atk: 31, def: 31, spa: 31, spd: 31, spe: 31};
+		}
```
Comment (urkerab) [inline data/random-teams.js]: Or I think `let ivs = Object.assign({hp: 31, atk: 31, def: 31, spa: 31, spd: 31, spe: 31}, setData.set.ivs);` also works.

File: data/random-teams.js [hunk 3]
```diff
@@ -1881,32 +1881,52 @@ class RandomTeams extends Dex.ModdedDex {
...
 			gender: setData.set.gender || template.gender || (this.randomChance(1, 2) ? 'M' : 'F'),
 			item: items + '' || setData.set.item || '',
 			ability: abilities + '' || setData.set.ability || template.abilities['0'],
-			shiny: typeof setData.set.shiny === 'undefined' ? this.randomChance(1, 1024) : setData.set.shiny,
+			shiny: setData.set.shiny || this.randomChance(1, 1024),
 			level: setData.set.level || 100,
-			happiness: typeof setData.set.happiness === 'undefined' ? 255 : setData.set.happiness,
-			evs: setData.set.evs || {hp: 84, atk: 84, def: 84, spa: 84, spd: 84, spe: 84},
-			ivs: setData.set.ivs || {hp: 31, atk: 31, def: 31, spa: 31, spd: 31, spe: 31},
+			happiness: setData.set.happiness || 255,
```
Comment (urkerab) [inline data/random-teams.js]: What happens for sets which specify `0` happiness?

Comment (KrisXV) [reply data/random-teams.js]: they have 0 happiness

File: data/random-teams.js (hunk 2 above)
Comment (KrisXV) [reply data/random-teams.js]: That's unnecessary

File: data/random-teams.js (hunk 1 above)
Comment (KrisXV) [reply data/random-teams.js]: no

File: data/random-teams.js (hunk 3 above)
Comment (Zarel) [reply data/random-teams.js]: Not under this code they don't.

**Comment by KrisXV** [comment]
ok done

File: data/random-teams.js (hunk 1 above)
Comment (Marty-D) [reply data/random-teams.js]: Might as well do the same thing with this, i.e. `Object.assign({hp: 0, atk: 0, def: 0, spa: 0, spd: 0, spe: 0}, setData.set.evs);` If there are EVs missing in future set updates it'll probably be easier to notice if they aren't 84 across the board instead.

**Comment by KrisXV** [comment]
done

File: data/random-teams.js [hunk 4]
```diff
@@ -1902,11 +1902,11 @@ class RandomTeams extends Dex.ModdedDex {
 			gender: setData.set.gender || template.gender || (this.randomChance(1, 2) ? 'M' : 'F'),
 			item: items + '' || setData.set.item || '',
 			ability: abilities + '' || setData.set.ability || template.abilities['0'],
-			shiny: typeof setData.set.shiny === 'undefined' ? this.randomChance(1, 1024) : setData.set.shiny,
-			level: setData.set.level || 100,
+			shiny: setData.set.shiny || this.randomChance(1, 1024),
```
Comment (Marty-D) [inline data/random-teams.js]: Should be changed back in case there's some situation where `"shiny": false` is important based on event moves or something?

**Comment by KrisXV** [comment]
done @Marty-D

**Pull request merged**